src/components/ProfileContent/ProfileContent.tsx

```diff
@@ -0,0 +1,225 @@
+
+type ProfileContentProps = {
+  isProfileInEditMode: boolean;
+  cancelEditClick: (prop: boolean) => void;
+};
+
+const ProfileContent = (props: ProfileContentProps) => {
+  const user = useAppSelector((state) => state.user);
+  const dispatch = useAppDispatch();
+  const [errorMsg, setErrorMsg] = useState<string>('');
+
+  const firstNameRef = useRef<HTMLInputElement>(null);
+  const countryRef = useRef<HTMLInputElement>(null);
+  const occupationRef = useRef<HTMLInputElement>(null);
+  const lastNameRef = useRef<HTMLInputElement>(null);
+  const birthdayRef = useRef<HTMLInputElement>(null);
+
+  const currentRefs =
+    firstNameRef.current &&
+    lastNameRef.current &&
+    occupationRef.current &&
+    countryRef.current &&
+    birthdayRef.current;
+
+  const cancelEditProfile = () => {
+    errorMsg && setErrorMsg('');
+    if (currentRefs) {
+      firstNameRef.current.value = user.firstName || '';
+      occupationRef.current.value = user.jobRole || '';
+      countryRef.current.value = '';
+      lastNameRef.current.value = user.lastName || '';
+      birthdayRef.current.value = '';
+    }
+    props.cancelEditClick(true);
+  };
+
+  const updateProfileFieldsDb = () => {
+    const userData = {
+      userId: user.userId,
+      newFields: { firstName: '', lastName: '', jobRole: '' },
+    };
+
+    if (currentRefs) {
+      const firstNameInputValue = firstNameRef.current.value;
+      const lastNameInputValue = lastNameRef.current.value;
+      const jobRoleInputValue = occupationRef.current.value;
+      const countryInputValue = countryRef.current.value;
+      const birthdayInputValue = birthdayRef.current.value;
+
+      const errMessage = validateEditProfile({
+        firstName: firstNameInputValue,
+        lastName: lastNameInputValue,
+        jobRole: jobRoleInputValue,
+        country: countryInputValue,
+        birthday: birthdayInputValue,
+      });
+      setErrorMsg(errMessage);
+
+      userData.newFields.firstName = firstNameInputValue;
+      userData.newFields.lastName = lastNameInputValue;
+      userData.newFields.jobRole = jobRoleInputValue;
+      if (errMessage) {
+        return;
+      }
+      birthdayRef.current.value = '';
+      countryRef.current.value = '';
+    }
+
+    dispatch(editUser(userData));
+  };
+
+  return (
+    <Container className="profile-content-container">
+      <Row className="first-row">
+        <Card className="card-content card-box">
+          <Card.Body className="card-content__body card-box__body">
+            <Card.Text className="card-content__text">
+              Some advice text.Some advice text.Some advice text.Some advice
+              text.Some advice text.Some advice text.Some advice text.Some
+              advice text.
+            </Card.Text>
+            <Card.Title className="card-content__title">Profile</Card.Title>
+          </Card.Body>
+        </Card>
+        <Col sm>
+          <Row>
+            <Col>First Name</Col>
```

Comment:
If you want, you can add * to the required fields, which are First and Last name, but Country, Birthday and Job role shouldn't be required, hence the user should be able to change and save only 1 of the fields. Also there isn't a validation for ' ' and I can change my names to just string with a space. 
![image](https://github.com/GenericSoft/GDR-Generic-Desk-Reservation/assets/31138395/30812cc8-3f65-432a-862c-d5069cd0c17b)